code/game/machinery/vending.dm

```diff
@@ -509,8 +509,6 @@
 			qdel(coin)
 			categories &= ~CAT_COIN
 
-	R.amount--

```

Comment:
May be worth flagging the machine as busy and stopping it accepting additional orders during the vending spawn() since it might be possible to cause it to vend below zero by spamming the interface during the delay. Either that or just put the first amount-- here and check if it's above zero before vending the second item.